core/math/vector2.h

```diff
@@ -185,6 +185,10 @@ struct [[nodiscard]] Vector2 {
 	operator Vector2i() const;
 
 	_FORCE_INLINE_ Vector2() {}
+	_FORCE_INLINE_ Vector2(real_t p) {
```

Comment:
Style: don't call the scalar parameter `p`.

The convention is that each parameter is formatted like `p_name` so you can distinguish it from fields that have no prefix. So prefer a name like `p_value` or `p_scalar` instead.
The same applies to the `variant_construct.cpp` changes.